### PR TITLE
feat: withdraw custom amount from savings

### PR DESCRIPTION
**Pull request by rolznz**
Closes https://github.com/getAlby/hub/issues/575

This allows users to send explicit amounts, e.g. to do a boltz swap, but is kept in the advanced options as Alby Hub is primarily a lightning wallet.

In the withdraw page, an input and send all checkbox are added. Because of how transaction fee calculation in bitcoin works, send all vs sending an amount are two different actions. The user is also warned if they send all when they have channels open, as it will drain their anchor reserves.

image

image

### CODE REVIEW

File: frontend/src/screens/wallet/WithdrawOnchainFunds.tsx [hunk 1]
```diff
@@ -139,18 +152,19 @@ export default function WithdrawOnchainFunds() {
...
-            <AlertDescription>
-              You have channels open and this withdrawal will use some or all of
-              your anchor reserves to publish the transaction, which may make it
-              harder to close channels without depositing additional onchain
-              funds to your savings balance.
-            </AlertDescription>
-          </Alert>
-        )}
+        {!!balances?.onchain.reserved &&
+          (sendAll || +amount > balances.onchain.total * 0.9) && (
```
Comment (rolznz) [inline frontend/src/screens/wallet/WithdrawOnchainFunds.tsx:156]: use spendable instead of total

**Comment by bumi** [comment]
btw. I find this screen a bit too complicated and I have never seen a wallet where I need to confirm my address and check a box.

why does it actually need to be that I am the owner of that address and having the seed phrase of it?

File: frontend/src/screens/wallet/WithdrawOnchainFunds.tsx [hunk 2]
```diff
@@ -159,6 +173,34 @@ export default function WithdrawOnchainFunds() {
...
+                />
+                <Label htmlFor="send-all" className="text-xs">
+                  Send All
+                </Label>
+              </div>
+            </div>
+            <Input
+              id="amount"
+              type="number"
+              value={sendAll ? balances.onchain.total : amount}
```
Comment (bumi) [inline frontend/src/screens/wallet/WithdrawOnchainFunds.tsx:196]: are we sure "send all" will be that amount?

Comment (rolznz) [reply frontend/src/screens/wallet/WithdrawOnchainFunds.tsx:196]: No, good catch!

**Comment by rolznz** [comment]
> btw. I find this screen a bit too complicated and I have never seen a wallet where I need to confirm my address and check a box.

@bumi this was because one of our users sent their funds to the wrong address :cry: 

I'll update the copy a bit.

**Comment by rolznz** [comment]
Updated UI:
image

image

Anchor reserve variants:

image

image

Replaced checkbox and confirm input with a confirmation dialog:

image

image

**Comment by bumi** [comment]
that's a nice UI.

**Pull request merged**